File: meal-planner-api/aspnet/MealPlanner.ObjectModels/Models/EatModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MealPlanner.ObjectModels.Models
{
  /// <summary>
  /// Represents the _Eat_ model (serving(s) eaten of a food item)
  /// </summary>
  public class EatModel : IValidatableObject
  {
    public AmountModel Servings { get; set; }
    public int FoodId { get; set; }

    /// <summary>
    /// Represents the _Eat_ `Validate` method
    /// </summary>
    /// <param name="validationContext"></param>
    /// <returns></returns>
    public IEnumerable<ValidationResult> Validate(ValidationContext validationContext) => new List<ValidationResult>();
  }
}

File: meal-planner-client/angular/src/app/data/eat.model.ts
import { ServingSize } from './serving-size.model';

/**
 * Represents the _Eat_ model
 *
 * ```yaml
 * servings: ServingSize;
 * foodId: number;
 * ```
 */
export interface Eat {
  servings: ServingSize;
  foodId: number;
}